shared/middleware/helpers/bucket_helper.rb

```diff
@@ -41,7 +45,7 @@ def cache_duration_seconds
   def app_size(encrypted_channel_id)
     owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
     prefix = s3_path owner_id, channel_id
-    @s3.list_objects(bucket: @bucket, prefix: prefix).contents.map(&:size).reduce(:+).to_i
+    s3.list_objects(bucket: @bucket, prefix: prefix).contents.map(&:size).reduce(:+).to_i
```

Comment:
Should this be `self.s3` throughout, or do we not need `self.s3` in the initialize method, or is this some ruby language weirdness I don't know about?

Reply:
it's ruby-language weirdness regarding getters vs setters:

`self.s3 = x` invokes `:s3=(x)` method, which is different from `s3 = x` setting a new local-variable named `s3` to the value `x`.

a getter isn't ambiguous in this way as long as no local-variable exists, so it can be invoked without the `self.` qualifier.

Reply:
:+1: Cool.  Also kind of weird... but I suspected it was something like this.